### ISSUE TITLE
Handle hyphenated words broken with a dash in ingredients lists (analysis and OCR extraction)

### ISSUE THREAD

**Issue by stephanegigandet** [openfoodfacts/openfoodfacts-server#3007]
We could try to "unbreak" words separated with a dash.

e.g. https://world.openfoodfacts.org/product/5411188118121/alpro

![image](https://user-images.githubusercontent.com/8158668/76195903-9008ee00-61e9-11ea-867d-baa3194ba853.png)

"eau, fèves de soja décort- quées (10,8%), citrate tricalcique, stabilisant (pectines), correcteurs d'acidité (citrates de sodium, acide citrique), arôme naturel, sel marin, antioxygènes (extrait riche en tocophé- rols, esters d'acides gras de l'acide ascor- bique), vitamines (B12, D2), ferments de yaourt (S. thermophilus, L. bulgaricus)."

Related to #2239

**Comment by hangy** [comment]
This could be pretty complex, as a hypen can also be a valid part of a word (at least in German). Basically, a dictionary would need to be integrated to avoid false positives.

**Comment by stephanegigandet** [comment]
@hangy : right, the way I do it for similar parsing features (like breaking "A and B" is to first check if the source "A and B" exists in the taxonomy, in that case I do nothing. If it doesn't exist, then I check if A exists and if B exists. If both exist, then I assume that it is 2 ingredients. We can try something similar for recombining.

**Comment by stephanegigandet** [comment]
" jus de pample - mousse rose à base de concentré 14%"  - https://fr.openfoodfacts.org/produit/3092719701443/sirop-de-pamplemousse-rose-zero-sucres-teisseire

**Comment by stephanegigandet** [comment]
From #3616 reported by @AcuarioCat in Spanish:

agua, habas de soja descascarilladas (4%), malto - dextrina (fibra), fructosa, azücar, car - bonato célcico, corrector de acidez (fosfato monopotésico), sal marina, aroma, estabilizante (goma gellan), vitaminas (riboflavina (B2), B12, D), aroma natural. Naturalmente sin lactosa y sin gluten.

which should be:

agua, habas de soja descascarilladas (4%), maltodextrina (fibra), fructosa, azücar, carbonato célcico, corrector de acidez (fosfato monopotésico), sal marina, aroma, estabilizante (goma gellan), vitaminas (riboflavina (B2), B12, D), aroma natural

**Comment by tuukka** [comment]
I'd like to work on this as I think it would be an important improvement.

@hangy In German specifically, I hope hyphens don't change the meaning of a word (even though they can make a word invalid), or do you have examples? In Finnish, I know it can happen in theory (classic example "luutaakka" vs. "luuta-akka") but probably not in practice within the ingredients vocabulary.
